libsel4muslcsys/src/sys_morecore.c

```diff
@@ -212,6 +213,8 @@ static long sys_mmap_impl_dynamic(void *addr, size_t length, int prot, int flags
         /* determine how many pages we need */
         uint32_t pages = BYTES_TO_4K_PAGES(length);
         void *ret = vspace_new_pages(muslc_this_vspace, seL4_AllRights, pages, seL4_PageBits);
+        ZF_LOGF_IF((((uintptr_t)ret) % 0x1000) != 0, "return address: 0x%lx requires alignment: 0x%lx ", (uintptr_t)ret,
```

Comment:
```suggestion
        ZF_LOGF_IF((((uintptr_t)ret) % 0x1000) != 0, "return address: %p requires alignment: 0x%x ", ret,
```
Just use "%p" for pointer. The additional "l" seems wrong for plain integers.

Reply:
It at least should be PRIxPTR

Reply:
fixed